pkg/metrics/chaos-controller-manager.go

```diff
@@ -0,0 +1,203 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package metrics
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/prometheus/client_golang/prometheus"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+
+	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+	"github.com/chaos-mesh/chaos-mesh/pkg/status"
+)
+
+var log = ctrl.Log.WithName("metrics-collector")
+
+// ChaosControllerManagerMetricsCollector implements prometheus.Collector interface
+type ChaosControllerManagerMetricsCollector struct {
+	store               cache.Cache
+	chaosExperiments    *prometheus.GaugeVec
+	SidecarTemplates    prometheus.Gauge
+	ConfigTemplates     *prometheus.GaugeVec
+	InjectionConfigs    *prometheus.GaugeVec
+	TemplateNotExist    *prometheus.CounterVec
+	TemplateLoadError   prometheus.Counter
+	ConfigNameDuplicate *prometheus.CounterVec
+	InjectRequired      *prometheus.CounterVec
+	Injections          *prometheus.CounterVec
+	chaosSchedules      *prometheus.GaugeVec
+	chaosWorkflows      *prometheus.GaugeVec
+}
+
+// NewChaosControllerManagerMetricsCollector initializes metrics and collector
+func NewChaosControllerManagerMetricsCollector(store cache.Cache, registerer prometheus.Registerer) *ChaosControllerManagerMetricsCollector {
+	c := &ChaosControllerManagerMetricsCollector{
+		store: store,
+		chaosExperiments: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "chaos_controller_manager_chaos_experiments",
+			Help: "Total number of chaos experiments and their phases",
+		}, []string{"namespace", "kind", "phase"}),
+		SidecarTemplates: prometheus.NewGauge(prometheus.GaugeOpts{
```

Comment:
I haven't been following this for a while. Is template and injections still used in chaos mesh?

Reply:
Yes. 

JVMChaos is still using this feature to inject an `initContainer` for preparing the javaagent. I'm not sure about the refined implementation of the new JVMChaos still needs this feature, but it is necessary to exist for a little while.

Reply:
> I'm not sure about the refined implementation of the new JVMChaos still needs this feature, but it is necessary to exist for a little while.

As I discussed with @WangXiangUSTC  earlier, the refined implementation of the new JVMChaos still needs this feature.

Reply:
> `-s` sets an access-all-areas security policy for the Byteman agent code. This may be necessary when the JVM is running an application employing a security manager which imposes access restrictions (this includes recent versions of JBoss Wildfly/EAP).

It seems byteman provides a flag `-s` for bypassing the JVM security policy, maybe useful. :D

ref: https://downloads.jboss.org/byteman/4.0.17/byteman-programmers-guide.html#installing-byteman-in-a-running-jvm-using-script-bminstall